plans/dht/test/common.go

```diff
@@ -389,23 +388,24 @@ func Bootstrap(ctx context.Context, runenv *runtime.RunEnv, watcher *sync.Watche
 
 // get all bootstrap peers.
 func getBootstrappers(ctx context.Context, runenv *runtime.RunEnv, watcher *sync.Watcher, opts *SetupOpts) ([]peer.AddrInfo, error) {
+	// cancel the sub
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Here for example we call `defer cancel()` right after creating the context - not consistent with how we use the context above with `watcher.Subscribe`.

Reply:
Should be functionally equivalent, but yeah, we should defer the cancel immediately above to avoid a goroutine leak: https://golang.org/pkg/context/#WithCancel.